tests/digest.bats

```diff
@@ -3,40 +3,92 @@
 load helpers
 
 fromreftest() {
-  _prefetch $1
-  run_buildah from --quiet --pull --signature-policy ${TESTSDIR}/policy.json $1
+  local img=$1
+
+  # Because the whalesay image flakes so often, give it three tries
+  run_buildah '?' from --quiet --pull --signature-policy ${TESTSDIR}/policy.json $img
+  if [ $status -ne 0 ]; then
+    echo "# pull failed; sleeping 5, then retry #1" >&3
+    sleep 5
+    run_buildah '?' from --quiet --pull --signature-policy ${TESTSDIR}/policy.json $img
+    if [ $status -ne 0 ]; then
+      echo "# pull failed again; sleeping 5, then retry #2" >&3
+      sleep 5
+      # This time, run without '?': if it fails, no more tries.
+      run_buildah from --quiet --pull --signature-policy ${TESTSDIR}/policy.json $img
+    fi
+  fi
   cid=$output
+
+  # If image includes '_v2sN', verify that image is schema version N
+  local expected_schemaversion=$(expr "$img" : '.*_v2s\([0-9]\)')
+  if [ -n "$expected_schemaversion" ]; then
+      actual_schemaversion=$(imgtype -expected-manifest-type '*' -show-manifest $img | jq .schemaVersion)
+      expect_output --from="$actual_schemaversion" "$expected_schemaversion" \
+                    ".schemaversion of $img"
+  fi
+
+  # This is all we test: basically, that buildah doesn't crash when pushing
   pushdir=${TESTDIR}/fromreftest
   mkdir -p ${pushdir}/{1,2,3}
-  run_buildah push --signature-policy ${TESTSDIR}/policy.json $1 dir:${pushdir}/1
+  run_buildah push --signature-policy ${TESTSDIR}/policy.json $img dir:${pushdir}/1
   run_buildah commit --signature-policy ${TESTSDIR}/policy.json $cid new-image
   run_buildah push --signature-policy ${TESTSDIR}/policy.json new-image dir:${pushdir}/2
   run_buildah rmi new-image
   run_buildah commit --signature-policy ${TESTSDIR}/policy.json $cid dir:${pushdir}/3
+
   run_buildah rm $cid
   rm -fr ${pushdir}
 }
 
 @test "from-by-digest-s1" {
-  fromreftest k8s.gcr.io/pause@sha256:bbeaef1d40778579b7b86543fe03e1ec041428a50d21f7a7b25630e357ec9247
+  fromreftest quay.io/libpod/testdigest_v2s1@sha256:816563225d7baae4782653efc9410579341754fe32cbe20f7600b39fc37d8ec7
 }
 
 @test "from-by-digest-s1-a-discarded-layer" {
-  fromreftest libpod/whalesay@sha256:2413c2ffc29fb01d51c27a91b804079995d6037eed9e4b632249fce8c8708eb4
+  # This image is responsible for a large number of test flakes, because it's
+  # huge and susceptible to network timeouts. Unfortunately, there doesn't
+  # seem to be anything we can do about it: the world has evolved since
+  # this image was created, and there no longer seem to be any tools that
+  # can create a similar image. (Yes, this means our testing is at the
+  # mercy of docker.io not removing that precise image from their registry)
+  IMG=docker.io/docker/whalesay@sha256:178598e51a26abbc958b8a2e48825c90bc22e641de3d31e18aaf55f3258ba93b
```

Comment:
should we snab a copy of the image and tuck it away in quay.io somwehere?

Reply:
regardless of the answer,  I wouldn't hold the merge back for this question.

Reply:
We can't: that's the entire point of my wasted few weeks of effort.

It seems like this image was magically written by earlier tools that (waves hands) did something that modern tools can no longer do. In other words: it is impossible to push this image anywhere and have it keep the properties that make it interesting. Current tools (buildah, skopeo, podman) somehow flatten/smoosh/deduplicate the layers on any new write. All we have is this image, on docker.io, and all we can do is hope it never changes.

Disclaimer: I have no idea what the technical terms are for any of this; I just know I've tried really really hard for several weeks and all my attempts fail on push.

Reply:
The current versions of our tools are dutifully cleaning up confusing bits of old data, which is rather inconvenient for us when we want to preserve those bits.

After some experimentation, it looks like having skopeo's vendored copy of containers/image not call `s1.fixManifestLayers()` in `github.com/containers/image/v5/manifest.Schema1FromManifest()` is enough to let it copy the image from docker hub to my test repository with its duplicate `blobSum` and `v1Compatibility` values intact.

I uploaded the change I made relative to [skopeo master](https://github.com/containers/skopeo/tree/e31d5a0e8fc61241b02b6eb06aeeb39ed4de7b5e) as a gist [here](https://gist.github.com/nalind/b491204ff05c3c3f3b6ef014b333a60c).  Hopefully it'll help preserve this detail for us.